### PR TITLE
Add linear drag

### PR DESCRIPTION
**Pull request by Thomio-Watanabe**
Several modifications in the drag force.

### CODE REVIEW

**Comment by doudou** [comment]
Hey. IMO, just get rid of the drag coefficients. They are really useless, we only want to have a single parameter multiplied by the velocity or velocity^2

**Comment by doudou** [comment]
finally, since we're changing the interface anyways, would you mind making the boyancy relative (i.e. say that the boyancy is 0.5 kg of lift instead of specifying it in absolute water displacement)

**Comment by Thomio-Watanabe** [comment]
Ok for the drag... I'll remove all drag coefficients and calculate it by the damp.
I didn't understand your comment about the buoyancy... could you be more specific.

**Comment by Thomio-Watanabe** [comment]
- Remove drag coefficients
- Add angular coefficients

File: src/GazeboUnderwater.cpp [hunk 1]
```diff
@@ -82,8 +92,14 @@ namespace gazebo_underwater
...
         densityOfFluid = getParameter<double>("fluid_density","kg/m3", 1027);
-        dragCoefficient = getParameter<math::Vector3>("drag_coefficient","dimensionless",
-                math::Vector3(1,1,1));
+        linearDampCoefficients = getParameter<math::Vector3>("linear_damp_coefficients","N.s/m",
+                math::Vector3(50,50,50));
+        linearDampAngleCoefficients = getParameter<math::Vector3>("linear_damp_coefficients","N.s",
+                math::Vector3(45,45,45));
+        quadraticDampCoefficients = getParameter<math::Vector3>("quadratic_damp_coefficients","N.s2/m2",
+                math::Vector3(40,40,40));
+        quadraticDampAngleCoefficients = getParameter<math::Vector3>("quadratic_damp_coefficients","N.s2/m",
```
Comment (doudou) [inline src/GazeboUnderwater.cpp]: shouldn't the parameter names be different ?

**Comment by Thomio-Watanabe** [comment]
Amend last commit. 
What about the buoyancy, should I remove the option to calculate the buoyancy using a given volume ? or you want the user to give the difference between the buoyancy and weight ?

**Comment by doudou** [comment]
> What about the buoyancy, should I remove the option to calculate the buoyancy using a given volume ? or you want the user to give the difference between the buoyancy and weight ?

I want the latter. The vehicles are basically designed to have some fixed amount of boyancy (between 500g and 1kg according to @quams).

**Comment by Thomio-Watanabe** [comment]
But the input difference between buoyancy and weight is given in kg (mass) instead of Newtons (force) ?

**Comment by doudou** [comment]
I would rather stick to newtons ...

**Comment by Thomio-Watanabe** [comment]
Get the difference between the buoyancy and weight

**Comment by Thomio-Watanabe** [comment]
Ping !

File: src/GazeboUnderwater.cpp [hunk 2]
```diff
@@ -81,12 +91,17 @@ namespace gazebo_underwater
...
+        linearDampCoefficients = getParameter<math::Vector3>("linear_damp_coefficients","N.s/m",
+                math::Vector3(50,50,50));
+        linearDampAngleCoefficients = getParameter<math::Vector3>("linear_damp_angle_coefficients","N.s",
+                math::Vector3(45,45,45));
+        quadraticDampCoefficients = getParameter<math::Vector3>("quadratic_damp_coefficients","N.s2/m2",
+                math::Vector3(40,40,40));
+        quadraticDampAngleCoefficients = getParameter<math::Vector3>("quadratic_damp_angle_coefficients","N.s2/m",
+                math::Vector3(35,35,35));
+        // buoyancy must be the difference between the buoyancy when the model is completely submersed and the model weight
+        buoyancy = getParameter<double>("buoyancy","N", 100);
```
Comment (doudou) [inline src/GazeboUnderwater.cpp]: That's 10kg of buoyancy .... A good baseline is 500g.

**Comment by Thomio-Watanabe** [comment]
- Modify default buoyancy value
  Ps: The simulation is breaking when the flatfish model is selected. (This might be related to the recent modifications in the model meshes !?)

**Comment by doudou** [comment]
> The simulation is breaking when the flatfish model is selected.

Breaking how ? Is that related to this PR ?

**Comment by Thomio-Watanabe** [comment]
- It is not related to this PR. I don't know which package is causing this issue.
- The simulation breaks when I click (select) the flat_fish model. The simulator closes without issuing a single warning message.
- I made a quick test with a box model, and the simulation also breaks when I select the box.

**Comment by Thomio-Watanabe** [comment]
I've just checked that the problem was related to the simulator itself (it had nothing to do with our packages). Then, after rebooting my PC it started to work properly.

**Comment by Thomio-Watanabe** [comment]
Ping !

**Pull request merged**